test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -330,6 +349,10 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
         namedWriteableRegistry = injector.getInstance(NamedWriteableRegistry.class);
     }
 
+    protected static void registerPlugin(Class<? extends Plugin> pluginClass) {
```

Comment:
I'll try out and see what the impact is. Before opening I was doubting if this should be done as I was unsure of the impact. But if it works out, this will end up being much cleaner.